wagtail/contrib/redirects/views.py

```diff
@@ -127,49 +127,24 @@ def header_more_buttons(self):
         ]
 
 
-@permission_checker.require("change")
-def edit(request, redirect_id):
-    theredirect = get_object_or_404(models.Redirect, id=redirect_id)
+class Edit(generic.EditView):
+    model = models.Redirect
+    form_class = RedirectForm
+    index_url_name = "wagtailredirects:index"
+    edit_url_name = "wagtailredirects:edit"
+    delete_url_name = "wagtailredirects:delete"
+    permission_required = "change_redirect"
+    delete_item_label = _("Delete")
+    submit_button_label = _("Save")
 
-    if not permission_policy.user_has_permission_for_instance(
-        request.user, "change", theredirect
-    ):
-        raise PermissionDenied
+    def get_object(self, queryset=None):
+        redirect_id = self.kwargs.get("redirect_id")
+        return get_object_or_404(self.model, id=redirect_id)
 
-    if request.method == "POST":
-        form = RedirectForm(request.POST, request.FILES, instance=theredirect)
-        if form.is_valid():
-            with transaction.atomic():
-                form.save()
-                log(instance=theredirect, action="wagtail.edit")
-            messages.success(
-                request,
-                _("Redirect '%(redirect_title)s' updated.")
-                % {"redirect_title": theredirect.title},
```

Comment:
This is one of those instances where I'm not really sure whether to remove the more specific string in favour of the generic one from the generic `EditView`. If we already have a more specific translatable string like this, I think it's good to keep it. It provides better context for translators compared to `"%(model_name)s '%(object)s' updated."`, which can be useful for languages with gendered nouns.

```suggestion
    def get_success_message(self):
        return _("Redirect '%(redirect_title)s' updated.") % {"redirect_title": self.object.title}
```